File: go/alb/alb.go
package alb

import (
	constant "cdk.tf/go/stack/constant"

	terraformelb "cdk.tf/go/stack/generated/hashicorp/aws/elb"
	"github.com/aws/jsii-runtime-go"
	"github.com/hashicorp/terraform-cdk-go/cdktf"
)

func CreateAlb(stack cdktf.TerraformStack) (cdktf.TerraformStack, terraformelb.Alb, terraformelb.AlbTargetGroup) {
	subnetsIds := []string{"subnet-10621875", "subnet-1c9cb65a"}

	newAlb := terraformelb.NewAlb(stack, jsii.String("aws_lb"), &terraformelb.AlbConfig{
		Name:             jsii.String(constant.Name),
		LoadBalancerType: jsii.String("application"),
		Subnets:          jsii.Strings(subnetsIds...),
	})

	newAlbTargetGroup := terraformelb.NewAlbTargetGroup(stack, jsii.String("aws_lb_target_group"), &terraformelb.AlbTargetGroupConfig{
		Name:       jsii.String(constant.Name),
		Port:       jsii.Number(4000),
		Protocol:   jsii.String(constant.Http),
		VpcId:      jsii.String("vpc-6f9b120a"),
		TargetType: jsii.String("ip"),
		HealthCheck: &terraformelb.AlbTargetGroupHealthCheck{
			Interval:           jsii.Number(70),
			Path:               jsii.String("/"),
			Port:               jsii.String("4000"),
			HealthyThreshold:   jsii.Number(2),
			UnhealthyThreshold: jsii.Number(2),
			Timeout:            jsii.Number(65),
			Protocol:           jsii.String(constant.Http),
			Matcher:            jsii.String("200"),
		},
	})

	terraformelb.NewLbListener(stack, jsii.String("aws_lb_listener"), &terraformelb.LbListenerConfig{
		Port:            jsii.Number(80),
		Protocol:        jsii.String(constant.Http),
		LoadBalancerArn: newAlb.Arn(),
		DefaultAction: []interface{}{
			map[string]interface{}{
				"type":           "forward",
				"targetGroupArn": *newAlbTargetGroup.Arn(),
			},
		},
	})

	return stack, newAlb, newAlbTargetGroup
}

File: typescript/construct/alb.ts
import { Construct } from 'constructs'
import { TerraformOutput } from 'cdktf'
import { elb } from '@cdktf/provider-aws'

export interface AlbConfig {
  readonly name: string,
  readonly subnetIds: string[],
  readonly vpcId: string
}

export class Alb extends Construct {
  public readonly lb: elb.Lb
  public readonly lbl: elb.LbListener
  public readonly lbTargetGroup: elb.LbTargetGroup

  constructor(scope: Construct, name: string, config: AlbConfig) {
    super(scope, name)

    this.lb = new elb.Lb(this, "appalb", {
      name: config.name,
      loadBalancerType: 'application',
      subnets: config.subnetIds
    })

    this.lbTargetGroup = new elb.LbTargetGroup(this, "app_target_group", {
      name: config.name,
      port: 4000,
      protocol: 'HTTP',
      vpcId: config.vpcId,
      deregistrationDelay: '100',
      targetType: 'ip',

      healthCheck: {
        interval: 70,
        path: '/',
        port: '4000',
        healthyThreshold: 2,
        timeout: 65,
        protocol: 'HTTP',
        matcher: '200'
      }
    })

    this.lbl = new elb.LbListener(this, "app_http_listener", {
      loadBalancerArn: this.lb.arn,
      port: 80,
      protocol: 'HTTP',

      defaultAction: [
        {
          type: 'forward',
          targetGroupArn: this.lbTargetGroup.arn
        }
      ]
    })

    new TerraformOutput(this, 'alb_dns', {
      value: this.lb.dnsName
    })
  }
}
